de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/48-de.metas.order/5552410_sys_gh6201_add_packing_item_validation_rule_for_invoiceLine.sql

```diff
@@ -0,0 +1,54 @@
+-- 2020-02-17T13:14:26.869Z
+-- I forgot to set the DICTIONARY_ID_COMMENTS System Configurator
+INSERT INTO AD_Val_Rule (AD_Client_ID,AD_Org_ID,AD_Val_Rule_ID,Code,Created,CreatedBy,EntityType,IsActive,Name,Type,Updated,UpdatedBy) VALUES (0,0,540480,'-- GENERAL
+M_HU_PI_Item_Product.AD_Org_ID IN (0, @AD_Org_ID@)
+AND (M_HU_PI_Item_Product.M_Product_ID=@M_Product_ID@ OR (M_HU_PI_Item_Product.isAllowAnyProduct=''Y'' AND M_HU_PI_Item_Product.M_HU_PI_Item_Product_ID not in (100)))
+AND ( M_HU_PI_Item_Product.C_BPartner_ID = @C_BPartner_ID@ OR M_HU_PI_Item_Product.C_BPartner_ID IS NULL)
+AND M_HU_PI_Item_Product.ValidFrom <= ''@DateInvoiced@'' AND ( ''@DateInvoiced@'' <= M_HU_PI_Item_Product.ValidTo OR M_HU_PI_Item_Product.ValidTo IS NULL )
+AND M_HU_PI_Item_Product.M_HU_PI_Item_ID IN
+	( SELECT i.M_HU_PI_Item_ID 
+	from M_HU_PI_Item i 
+	where i.M_HU_PI_Version_ID IN
+		(SELECT v.M_HU_PI_Version_ID 
+		from M_HU_PI_Version v 
+		where v.HU_UnitType = ''TU''
+		)
+	)
+	
+-- Price list
+AND
+(
+	-- AUFTRAG
+	(
+		''@IsSOTrx@'' = ''Y''
+	)
+
+	-- BESTELLUNG
+	OR
+	(
+		''@IsSOTrx@'' = ''N''
```

Comment:
Why do we need all this pricing stuff this only for purchase invoices and not for sales invoices?
Spoiler: we do need it for both :-) !

Background:
in sales orders, the packaging is decided on shipment-time. So the sales order does not have to worry about what packaging will be shippend and invoiced later. Only for purchase orders we need the packaging products' prices on onrder time.

However AFAIU
the invoice line *does* need to worry if the invoiced packaing product has a price!